From 9d5bec80895231fe5575f479cc2fcaf4735daff1 Mon Sep 17 00:00:00 2001
From: Eric Lundby <Eric.Lundby@gmail.com>
Date: Wed, 8 Oct 2025 17:28:01 -0600
Subject: [PATCH 1/1] use-conda-env

- Make sure the build is configured for Release.
- Pass -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreadedDLL to match llvmdev.
- Pass the env to the build processes.

---
 ffi/build.py | 27 +++++++++++++++++++++++----
 1 file changed, 23 insertions(+), 4 deletions(-)

diff --git a/ffi/build.py b/ffi/build.py
index 779471e..972d288 100755
--- a/ffi/build.py
+++ b/ffi/build.py
@@ -89,10 +89,14 @@ def try_cmake(cmake_dir, build_dir, generator, arch=None, toolkit=None):
     args.append(cmake_dir)
     cmake_options = env_var_options_to_cmake_options()
     args += cmake_options
+    args += [
+        "-DCMAKE_BUILD_TYPE=Release",
+        "-DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreadedDLL",
+    ]
     try:
         os.chdir(build_dir)
         print('Running:', ' '.join(args))
-        subprocess.check_call(args)
+        subprocess.check_call(args, env=os.environ)
     finally:
         os.chdir(old_dir)
 
@@ -166,8 +170,23 @@ def main_windows():
         os.mkdir(build_dir)
     # Run configuration step
     try_cmake(here_dir, build_dir, *generator)
-    subprocess.check_call(['cmake', '--build', build_dir, '--config', config])
-    shutil.copy(os.path.join(build_dir, config, 'llvmlite.dll'), target_dir)
+    subprocess.check_call(['cmake', '--build', build_dir, '--config', config], env=os.environ)
+    
+    # Find the DLL file - it might be in different locations depending on generator
+    dll_paths = [
+        os.path.join(build_dir, config, 'llvmlite.dll'),  # Visual Studio
+        os.path.join(build_dir, 'llvmlite.dll'),          # Ninja
+    ]
+    
+    dll_found = False
+    for dll_path in dll_paths:
+        if os.path.exists(dll_path):
+            shutil.copy(dll_path, target_dir)
+            dll_found = True
+            break
+    
+    if not dll_found:
+        raise RuntimeError(f"Could not find llvmlite.dll in any expected location: {dll_paths}")
 
 
 def main_posix(library_ext):
@@ -177,7 +196,7 @@ def main_posix(library_ext):
         os.mkdir(build_dir)
     try_cmake(here_dir, build_dir, generator)
     cmd = ['cmake', '--build', build_dir, "--parallel", '--config', config]
-    subprocess.check_call(cmd)
+    subprocess.check_call(cmd, env=os.environ)
     shutil.copy(os.path.join(build_dir, 'libllvmlite' + library_ext),
                 target_dir)
 
-- 
2.45.2

